scripts/download-e2e-binaries.sh

```diff
@@ -35,7 +35,7 @@ dest_dir="${root_dir}/bin"
 mkdir -p "${dest_dir}"
 
 # kind
-kind_version="v0.4.0"
+kind_version="v0.7.0"
```

Comment:
perhaps we could try using https://github.com/kubernetes-sigs/kind/releases/tag/v0.8.1

Reply:
I was thinking about that, too, but then refrained from it because I guess that the adoption rate of 0.8.x is still pretty low. wdyt?

The best solution would be to iterate through multiple versions, like 0.7.x and 0.8.x but I'd prefer to do that in another PR. :-)

Reply:
Plus: v0.8.0 introduced some breaking changes that I'd have to adapt to additional to the changes I'm already tackling in this PR. So it seems to be a good idea to go for 0.7 for now.